pkg/controllerutils/mapper/mapper.go

```diff
@@ -53,9 +52,8 @@ func (m *clusterToObjectMapper) Map(ctx context.Context, _ logr.Logger, reader c
 
 // ClusterToObjectMapper returns a mapper that returns requests for objects whose
 // referenced clusters have been modified.
-func ClusterToObjectMapper(ctx context.Context, mgr manager.Manager, newObjListFunc func() client.ObjectList, predicates []predicate.Predicate) Mapper {
+func ClusterToObjectMapper(mgr manager.Manager, newObjListFunc func() client.ObjectList, predicates []predicate.Predicate) Mapper {
```

Comment:
Thanks! When I was vendoring to gardener/gardener@v1.66 in the registry-cache extension I was wondering whether we can pass the dedicated needed objs instead of passing the whole `mgr manager.Manager`.
For the `ClusterToObjectMapper` only the mgr's cache is used for example. I think this can make the things easier when writing unit tests. You don't need to pass the whole `manager.Manager` but only the things you need - client, decoder, and whether else is needed.
Is there a convention to pass the whole manager?
If you want, we can also leave this for a dedicated PR as it is not strictly related to the content from this PR.

cc @shafeeqes

Reply:
There is no convention as such, It was discussed in https://github.com/gardener/gardener/pull/8217#discussion_r1257710092, but for streamlining reasons, and if we need more things from the manager in the future, it was decided to pass the whole manager.
cc @ary1992